packages/serverless/src/domain/use-case/user.ts

```diff
@@ -3,83 +3,109 @@ import {
   TPagination,
   TPaginationReturn,
   TUser,
+  TUserRecord,
 } from '@zkdb/common';
 import Client from 'mina-signer';
-import { ClientSession, FindOptions } from 'mongodb';
+import { ClientSession, WithoutId } from 'mongodb';
 import { DEFAULT_PAGINATION } from '../../common/const.js';
+import { getCurrentTime } from '../../helper/common.js';
 import config from '../../helper/config.js';
-import logger from '../../helper/logger.js';
 import ModelUser from '../../model/global/user.js';
 import { FilterCriteria } from '../utils/document.js';
 
-export async function findUser(
-  query?: FilterCriteria,
-  paginationInput?: TPagination,
-  session?: ClientSession
-): Promise<TPaginationReturn<TUser[]>> {
-  const modelUser = new ModelUser();
+export type TUserParamSignUp = {
+  // Remove publicKey from TUser since TMinaSignature already have
+  user: Omit<TUser, 'publicKey'>;
+  signature: TMinaSignature;
+};
 
-  const options: FindOptions = {};
-  const pagination = paginationInput || DEFAULT_PAGINATION;
+export type TUserParamFindPagination = {
+  query?: FilterCriteria;
+  paginationInput?: TPagination;
+};
 
-  return {
-    data: await modelUser
-      .find(query || {}, {
-        session,
-        ...options,
-      })
-      .toArray(),
-    ...pagination,
-    total: await modelUser.count(query, { session }),
-  };
-}
-
-export async function isUserExist(userName: string): Promise<boolean> {
-  const modelUser = new ModelUser();
-  return (await modelUser.findOne({ userName })) !== null;
-}
-
-export async function signUpUser(user: TUser, signature: TMinaSignature) {
-  const { userName, publicKey, email } = user;
-  const client = new Client({ network: config.NETWORK_ID });
-  if (client.verifyMessage(signature)) {
-    const jsonData: TUser = JSON.parse(signature.data);
-    if (jsonData.userName !== userName) {
-      throw new Error('Username does not match');
-    }
-    if (jsonData.email !== email) {
-      throw new Error('Email does not match');
-    }
-    const modelUser = new ModelUser();
+export class User {
+  public static async signUp(
+    params: TUserParamSignUp
```

Comment:
Check my comments on `params` in `Group` then apply for `User` as well.

Reply:
I change to `paramSignUp` since the `signUp` param is special and can't be reuse. Also I move type to common